src/test/java/org/broadinstitute/hellbender/utils/UtilsUnitTest.java

```diff
@@ -731,4 +731,84 @@ public void testGetDuplicatedItems(final Collection<?> collection, final Set<?>
+                    new Object[] { stringData, "",    Arrays.asList( stringData.split(""   ) ) },
+                    new Object[] { stringData, "a",   Arrays.asList( stringData.split("a"  ) ) },
+                    new Object[] { stringData, "b",   Arrays.asList( stringData.split("b"  ) ) },
+                    new Object[] { stringData, "c",   Arrays.asList( stringData.split("c"  ) ) },
+                    new Object[] { stringData, "d",   Arrays.asList( stringData.split("d"  ) ) },
+                    new Object[] { stringData, "e",   Arrays.asList( stringData.split("e"  ) ) },
+                    new Object[] { stringData, "f",   Arrays.asList( stringData.split("f"  ) ) },
+                    new Object[] { stringData, "g",   Arrays.asList( stringData.split("g"  ) ) },
+                    new Object[] { stringData, "h",   Arrays.asList( stringData.split("h"  ) ) },
+                    new Object[] { stringData, "i",   Arrays.asList( stringData.split("i"  ) ) },
+                    new Object[] { stringData, "j",   Arrays.asList( stringData.split("j"  ) ) },
+                    new Object[] { stringData, "k",   Arrays.asList( stringData.split("k"  ) ) },
+                    new Object[] { stringData, "l",   Arrays.asList( stringData.split("l"  ) ) },
+                    new Object[] { stringData, "m",   Arrays.asList( stringData.split("m"  ) ) },
+                    new Object[] { stringData, "n",   Arrays.asList( stringData.split("n"  ) ) },
+                    new Object[] { stringData, "o",   Arrays.asList( stringData.split("o"  ) ) },
+                    new Object[] { stringData, "p",   Arrays.asList( stringData.split("p"  ) ) },
+                    new Object[] { stringData, "q",   Arrays.asList( stringData.split("q"  ) ) },
+                    new Object[] { stringData, "r",   Arrays.asList( stringData.split("r"  ) ) },
+                    new Object[] { stringData, "s",   Arrays.asList( stringData.split("s"  ) ) },
+                    new Object[] { stringData, "t",   Arrays.asList( stringData.split("t"  ) ) },
+                    new Object[] { stringData, "u",   Arrays.asList( stringData.split("u"  ) ) },
+                    new Object[] { stringData, "v",   Arrays.asList( stringData.split("v"  ) ) },
+                    new Object[] { stringData, "w",   Arrays.asList( stringData.split("w"  ) ) },
+                    new Object[] { stringData, "x",   Arrays.asList( stringData.split("x"  ) ) },
+                    new Object[] { stringData, "y",   Arrays.asList( stringData.split("y"  ) ) },
+                    new Object[] { stringData, "z",   Arrays.asList( stringData.split("z"  ) ) },
+                    new Object[] { stringData, " ",   Arrays.asList( stringData.split(" "  ) ) },
+                    new Object[] { stringData, "T",   Arrays.asList( stringData.split("T"  ) ) },
+                    new Object[] { stringData, "9",   Arrays.asList( stringData.split("9") ) }
```

Comment:
You are still missing a number of important edge cases in your test data. Eg.,

-A String consisting only of delimiters: ":::::" with a delimiter of ":"
-Multiple consecutive delimiters at the beginning, middle, and/or end of the String ("::foo", "foo::", "foo::bar" with a delimiter of ":")

Try to enumerate all meaningfully-different arrangements of delimiter and non-delimiter in short Strings.

Reply:
I added a few more test cases that I think handle this.